runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodePlugin.java

```diff
@@ -0,0 +1,134 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.motherlode;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Binder;
+import com.google.inject.Provides;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.task.Schedule;
+import net.runelite.client.ui.overlay.Overlay;
+
+import javax.imageio.ImageIO;
+import javax.inject.Inject;
+import java.awt.image.BufferedImage;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Collection;
+
+@PluginDescriptor(
+	name = "Motherlode plugin"
+)
+public class MotherlodePlugin extends Plugin
+{
+	private static BufferedImage mineIcon;
+
+	@Inject
+	MotherlodeOverlay overlay;
+
+	@Inject
+	MotherlodeRocksOverlay rocksOverlay;
+	@Inject
+	MotherlodeSackOverlay motherlodeSackOverlay;
+
+	@Inject
+	MotherlodeConfig config;
+
+	private final MotherlodeSession session = new MotherlodeSession();
+
+	@Override
+	public void configure(Binder binder)
+	{
+		binder.bind(MotherlodeOverlay.class);
+	}
+
+	@Provides
+	MotherlodeConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(MotherlodeConfig.class);
+	}
+
+	@Override
+	public Collection<Overlay> getOverlays()
+	{
+		return Arrays.asList(overlay, rocksOverlay, motherlodeSackOverlay);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		mineIcon = ImageIO.read(MotherlodePlugin.class.getResourceAsStream("/skill_icons/mining.png"));
+	}
+
+	public MotherlodeSession getSession()
+	{
+		return session;
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getType() == ChatMessageType.FILTERED)
+		{
+			if (event.getMessage().equals("You manage to mine some pay-dirt."))
+			{
+				session.incrementPayDirtMined();
+			}
+		}
+	}
+
+	@Schedule(
+		period = 1,
+		unit = ChronoUnit.SECONDS
+	)
+	public void checkMining()
+	{
+		Instant lastPayDirtMined = session.getLastPayDirtMined();
+		if (lastPayDirtMined == null)
+		{
+			return;
+		}
+
+		// reset recentPayDirtMined if you haven't mined anything recently
+		Duration statTimeout = Duration.ofMinutes(config.statTimeout());
+		Duration sinceMined = Duration.between(lastPayDirtMined, Instant.now());
+
+		if (sinceMined.compareTo(statTimeout) >= 0)
+		{
+			session.resetRecent();
+		}
+	}
+
+	public static BufferedImage getMineIcon()
```

Comment:
This should be instance method, or mineIcon should be initialized in static context.